CHANGELOG.md

```diff
@@ -32,6 +38,57 @@ NB: this is not necessary if a user has `READONLY = 2` mode as it allows to modi
 
 See also: [readonly documentation](https://clickhouse.com/docs/en/operations/settings/permissions-for-queries#readonly).
 
+### New features
+
+- Added `url` configuration parameter. It is intended to replace the deprecated `host`, which was already supposed to be passed a valid URL.
+- It is now possible to configure most of the client instance parameters with a URL. The URL format is `http[s]://[username:password@]hostname:port[/database][?param1=value1&param2=value2]`. The name of a particular parameter is supposed to reflect its path in the config options interface. The following parameters are supported:
+
+  - `readonly` - boolean. See below [1].
+  - `application_id` - non-empty string.
+  - `session_id` - non-empty string.
+  - `request_timeout` - non-negative number.
+  - `max_open_connections` - non-negative number, greater than zero.
+  - `compression_request` - boolean.
+  - `compression_response` - boolean.
+  - `log_level` - allowed values: `OFF`, `TRACE`, `DEBUG`, `INFO`, `WARN`, `ERROR`.
+  - `keep_alive_enabled` - boolean.
+  - `clickhouse_settings_*` - see below [2].
```

Comment:
Nit: maybe `clickhouse_setting_*` since it's a single setting? a side note: python client uses `ch_` convention https://github.com/ClickHouse/clickhouse-connect/blob/01dcaf7c3494614e6c773ee10f744ff475d5b26e/clickhouse_connect/driver/__init__.py#L104

Reply:
My idea was that it is supposed to reflect the path in the settings. We can add a shorthand, too, i.e., you can write `clickhouse_settings_` or just `ch_`